kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMDeviceStateTest.java

```diff
@@ -0,0 +1,203 @@
+		whenInt32StateIsPassed(new UInt32(110));
+		thenStateShouldBeEqualTo(NMDeviceState.NM_DEVICE_STATE_DEACTIVATING);
+	}
+
+	@Test
+	public void conversionWorksForStateFailed() {
+		whenInt32StateIsPassed(new UInt32(120));
+		thenStateShouldBeEqualTo(NMDeviceState.NM_DEVICE_STATE_FAILED);
+	}
+
+	@Test
+	public void conversionWorksForStateUnknownDefault() {
+		whenInt32StateIsPassed(new UInt32(121));
+		thenStateShouldBeEqualTo(NMDeviceState.NM_DEVICE_STATE_UNKNOWN);
+	}
+
+	@Test
+	public void shouldReturnConnectionFalseUnknown() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_UNKNOWN);
+		thenStateShouldBeEqualTo(false);
+	}
+
+	@Test
+	public void shouldReturnConnectionFalseUnmanaged() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_UNMANAGED);
+		thenStateShouldBeEqualTo(false);
+	}
+
+	@Test
+	public void shouldReturnConnectionFalseUnavailable() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_UNAVAILABLE);
+		thenStateShouldBeEqualTo(false);
+	}
+
+	@Test
+	public void shouldReturnConnectionFalseDisconnected() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_DISCONNECTED);
+		thenStateShouldBeEqualTo(false);
+	}
+
+	@Test
+	public void shouldReturnConnectionTruePrepare() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_PREPARE);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	@Test
+	public void shouldReturnConnectionTrueConfig() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_CONFIG);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	@Test
+	public void shouldReturnConnectionTrueNeedAuth() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_NEED_AUTH);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	@Test
+	public void shouldReturnConnectionTrueIpConfig() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_IP_CONFIG);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	@Test
+	public void shouldReturnConnectionTrueIpCheck() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_IP_CHECK);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	@Test
+	public void shouldReturnConnectionTrueSecondaries() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_SECONDARIES);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	@Test
+	public void shouldReturnConnectionTrueActivated() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_ACTIVATED);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	@Test
+	public void shouldReturnConnectionTrueDeactivating() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_DEACTIVATING);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	@Test
+	public void shouldReturnConnectionTrueFailed() {
+		whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_FAILED);
+		thenStateShouldBeEqualTo(true);
+	}
+
+	public void whenInt32StateIsPassed(UInt32 type) {
+		state = NMDeviceState.fromUInt32(type);
+	}
+
+	public void whenStateIsSetTo(NMDeviceState type) {
+		state = type;
+	}
+
+	public void thenStateShouldBeEqualTo(NMDeviceState type) {
+		assertEquals(state, type);
+	}
+
+	public void thenStateShouldBeEqualTo(Boolean bool) {
```

Comment:
The previous name was better:

```suggestion
	public void thenIsConnectShouldReturn(Boolean bool) {
```